cmd/osctl/pkg/client/config/path.go

```diff
@@ -0,0 +1,40 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package config
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/talos-systems/talos/pkg/constants"
+)
+
+// GetTalosDirectory returns path to Talos directory (~/.talos by default).
+func GetTalosDirectory() (string, error) {
+	if talosDir, ok := os.LookupEnv(constants.TalosDirectoryEnvVar); ok {
+		return talosDir, nil
+	}
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(home, ".talos"), nil
```

Comment:
I'm torn on the `TALOSDIR` env var. I would prefer an explicit flag. Kinda like how `containerd` functions (it has a flag for specifying where to store state). Configurability is nice, but I think we should keep this particular option as simple and opinionated as possible. Knowing that `~/.talos` is a directory we "own" is easier to communicate and makes for less moving parts. Configuring this via a flag would be clear and explicit still.

Reply:
Explicit flag is there for `~/.talos/clusters`, but this provides default for `~/.talos` itself (and influences config location). I can hardcode this to `~/.talos` though.